### DISCUSSION TITLE
debian executable issue due to snap7

### DISCUSSION THREAD

**Discussion by Diptee24** [gijzelaerr/python-snap7#421] [Q&A] [answered]
i made executable using pyinstaller and my application use snap 7 library.
after generating executable successfull when run executable it give snap7 error.

**Comment by gijzelaerr** [comment]
What is the error

**Comment by Diptee24** [comment]
can't find snap7 library. If installed, try running ldconfig
but i have already installed using sudo python -m pip install python-snap7 and i can run using terminal

**Comment by spreeker** [comment]
python-snap7 is a wrapper around a C library called snap7. that C library is apparently missing. read documentation.

**Comment by Diptee24** [comment]
but it works from terminal and from where i can get this file and where i have to put

**Comment by gijzelaerr** [comment]
you need to figure out how to bundle the snap7 shared libraries into your pyinstaller package, that is out of scope for python-snap7.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as resolved. This was a PyInstaller bundling issue with the native snap7 library. Python-snap7 3.0 is now a pure Python implementation, so native library bundling is no longer needed.
